internal/common/context.bzl

```diff
@@ -0,0 +1,328 @@
+load("//internal/common/actions:actions.bzl", "create_source_dir", "run_js")
+load("//internal/npm_packages:rule.bzl", "NpmPackagesInfo")
+# load("@rules_nodejs//internal/common:node_module_info.bzl", "NodeModuleInfo", "collect_node_modules_aspect")
+# load("@rules_nodejs//internal/common:module_mappings.bzl", "get_module_mappings", "module_mappings_runtime_aspect")
+
+###############################################################################
+# Providers
+
+# The Providers used are inspired by the rules_go providers:
+# https://github.com/bazelbuild/rules_go/blob/master/go/providers.rst#GoLibrary
+
+# The NodeModuleInfo is provided by the rules_nodejs and generated by the
+# collect_node_modules_aspect when the target has a "tags" attribute with
+# "NODE_MODULE_MARKER" in it.
+# NodeModuleInfo = provider(
+#   fields = {
+# Only the workspace is needed because rules_
```

Comment:
Is the sentence incomplete here?

Reply:
Yeah, I should take it out. I don't think that we will be going down the NODE_MODULE_MARKER route. Although if I got some spare time I might implement it to get these rules to integrate with `rulse_nodejs` and `rules_typescript`.